misc/multiaddr/src/protocol.rs

```diff
@@ -50,7 +51,7 @@ pub enum Protocol<'a> {
     P2pWebRtcStar,
     P2pWebSocketStar,
     Memory,
-    Onion(Cow<'a, [u8]>),
+    Onion(Cow<'a, [u8; 12]>),
```

Comment:
If I understand correctly, this `[u8; 12]` contains 10 bytes of the address and 2 bytes of port.
What about having two fields instead?